data/routers/project-router.js

```diff
@@ -43,69 +49,88 @@ router.post('/', async (req, res) => {
 });
 
 router.put('/:id', async (req, res) => {
-  if (!req.params.id || !req.body.changes) {
+  const { name, description, completed } = req.body;
+  if (!name && !description && !completed) {
     res.status(400).json({
-      error:
-        'Please provide the ID of the project you intend to update as well as your intended changes.'
+      error: 'Please provide the project changes you intend to make.'
     });
   } else {
     try {
-      const updatedProject = await projectModel.update(
-        req.params.id,
-        req.body.changes
-      );
+      const updatedProject = await projectModel.update(req.params.id, req.body);
       if (updatedProject) {
-        res
-          .status(200)
-          .json({ message: 'You successfully updated the project.' });
+        res.status(200).json(updatedProject);
       } else {
         res.status(404).json({
-          message: 'The project with the specified ID does not exist.'
+          message: 'There is no project with the specified ID.'
         });
       }
     } catch (err) {
       res.status(500).json({
-        error: 'There was an error while updating the project.'
+        error: `There was an error while updating the project. ${err}`
       });
     }
   }
 });
 
 router.delete('/:id', async (req, res) => {
   try {
-    const numOfRemovedProjects = await projectModel.remove(req.params.id);
-    if (numOfRemovedProjects) {
-      res.status(200).json({
-        message: `Number of projects deleted: ${numOfRemovedProjects}`
-      });
+    const project = await projectModel.get(req.params.id);
+    if (project) {
+      try {
+        const projectActions = await projectModel.getProjectActions(
+          req.params.id
+        );
+        if (projectActions.length > 0) {
+          return res.status(400).json({
+            error:
+              "You must delete all a project's actions before deleting the project."
+          });
```

Comment:
based on how the databases were set up, yes, but not that the helpers behind the scenes should have deleted a project's actions when attempting to delete that project for you

Reply:
My testing seemed to show that a project with actions could be deleted while leaving project-less actions hanging around.